src/main/java/htsjdk/samtools/filter/IntervalFilter.java

```diff
@@ -94,6 +94,9 @@ private void advanceInterval() {
      * @return true if the SAMRecords matches the filter, otherwise false
      */
     public boolean filterOut(final SAMRecord first, final SAMRecord second) {
-        throw new UnsupportedOperationException("Paired IntervalFilter filter not implemented!");
+        // This can never be implemented because if the bam is coordinate sorted,
+        // which it has to be for this filter, it will never get both the first and second reads together
+        // and the filterOut method goes in order of the intervals in coordinate order so it will miss reads.
+        throw new UnsupportedOperationException("Paired IntervalFilter filter cannot be implemented, use IntervalKeepPairFilter.");
```

Comment:
I would use `IntervalKeepPairFilter` but `IntervalKeepPairFilter` doesn't support this method ;)